### ISSUE TITLE
Java(jmod) support?

### ISSUE THREAD

**Issue by chengenzhao** [gfx-rs/gfx#2425]
Short info header:
- GFX version:
latest
- OS:
all
- GPU:
all

Hi, recently Java provide a new way to generate jre, which is jmod
http://openjdk.java.net/jeps/261
and is it possible to provide jmod format support？
thus all jvm languages could use gfx-rs to create games
I think the way to do this is similar to JavaFX project
https://github.com/javafxports/openjdk-jfx

**Comment by kvark** [comment]
I have no idea what you are talking about. Can you provide a gist of the links involved?
Can java link with Rust/C libraries now? Or can it compile with LLVM?

**Comment by chengenzhao** [comment]
> I have no idea what you are talking about. Can you provide a gist of the links involved?
> Can java link with Rust/C libraries now? Or can it compile with LLVM?

Hi, yes, Java could link Rust&C libs now, so basically the way to do it is like this:
https://github.com/jni-rs/jni-rs
tutorial: https://docs.rs/jni/0.10.2/jni/
so basically what we suggest is to expose gfx-rs api to jvm
and then wrap it as a jmod using jmod command in java/home/bin/ directory
thus when we wants to use it, 
we could create our own runtime and put rust crate(gfx-rs) in the runtime
and then all other languages including Java could use gfx-rs as its own api

The other way is to use graal which could transform llvm bitcode into bytecode but I think that way is a little bit slow since there are transformations in the runtime

**Comment by kvark** [comment]
@chengenzhao this is very interesting, thank you for the info! I think it would be great to have, but the only way to get there is to have somebody (knowledgeable of Java) to lead the way.

**Comment by kvark** [comment]
@vitvakatu good point! If Java has any access to Vulkan, they can just run with gfx-portability.
